app/controllers/users_controller.rb

```diff
@@ -16,7 +16,11 @@ def create
   end
 
   def show
-    @user = User.find(session[:user_id])
+    if session[:user_id].nil?
+      render file: "/public/404"
```

Comment:
@Dpalazzari @njgheorghita Wouldn't a redirect to `root_path` with a flash message to the user would be more user friendly for this use case? Can you please help to confirm why showing 404 page is done here? FYI @drod1000